mainargs/src-3/Macros.scala

```diff
@@ -0,0 +1,155 @@
+package mainargs
+
+import scala.quoted._
+
+object Macros {
+  private def mainAnnotation(using Quotes) = quotes.reflect.TypeRepr.of[mainargs.main].typeSymbol
+  private def argAnnotation(using Quotes) = quotes.reflect.TypeRepr.of[mainargs.arg].typeSymbol
+  def parserForMethods[B](base: Expr[B])(using Quotes, Type[B]): Expr[ParserForMethods[B]] = {
+    import quotes.reflect._
+    val allMethods = TypeRepr.of[B].typeSymbol.memberMethods
+    val annotatedMethodsWithMainAnnotations = allMethods.flatMap { methodSymbol =>
+      methodSymbol.getAnnotation(mainAnnotation).map(methodSymbol -> _)
+    }.sortBy(_._1.pos.map(_.start))
+    val mainDatas = Expr.ofList(annotatedMethodsWithMainAnnotations.map { (annotatedMethod, mainAnnotationInstance) =>
+      createMainData[Any, B](annotatedMethod, mainAnnotationInstance)
+    })
+
+    '{
+      new ParserForMethods[B](
+        MethodMains[B](${ mainDatas }, () => ${ base })
+      )
+    }
+  }
+
+  def parserForClass[B](using Quotes, Type[B]): Expr[ParserForClass[B]] = {
+    import quotes.reflect._
+    val typeReprOfB = TypeRepr.of[B]
+    val companionModule = typeReprOfB match {
```

Comment:
Is there a chance that `B` doesn't have a companion module?
If so then a graceful fail might be needed here (or if `typeSymbolOfB` is `NoSymbol`)

Reply:
>  If so then a graceful fail might be needed here (or if typeSymbolOfB is NoSymbol)

Do you know how? `typeSymbolOfB` is not undefined. But the compiler crashes in a way I cannot understand 🤔 

```

#2] [info] exception occurred while compiling /Users/lorenzo/scala/mainargs/mainargs/test/src-3/VersionSpecific.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/Checker.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/ClassTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/CoreTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/DashedArgumentName.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/FlagTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/HelloWorldTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/HygieneTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/MacroTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/ManyTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/NewVarargsTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/OptionSeqTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/ParserTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/PositionalTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/TestUtils.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/VarargsTests.scala
[#0] scala.MatchError: Ident(apply) (of class dotty.tools.dotc.ast.Trees$Ident) while compiling /Users/lorenzo/scala/mainargs/mainargs/test/src-3/VersionSpecific.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/Checker.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/ClassTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/CoreTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/DashedArgumentName.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/FlagTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/HelloWorldTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/HygieneTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/MacroTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/ManyTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/NewVarargsTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/OptionSeqTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/ParserTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/PositionalTests.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/TestUtils.scala, /Users/lorenzo/scala/mainargs/mainargs/test/src/VarargsTests.scala
[#2] [error] ## Exception when compiling 16 sources to /Users/lorenzo/scala/mainargs/out/mainargs/jvm/3.0.2/test/compile.dest/classes
[#2] [error] scala.MatchError: Ident(apply) (of class dotty.tools.dotc.ast.Trees$Ident)
[#2] [error] dotty.tools.dotc.typer.Typer.newExpr$1(Typer.scala:3759)
[#2] [error] dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:3804)
[#2] [error] dotty.tools.dotc.typer.Typer.adapt(Typer.scala:3162)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2941)
[#2] [error] dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:882)
[#2] [error] dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1032)
[#2] [error] dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:317)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedApply(Inliner.scala:1535)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2693)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2756)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2818)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.Typer.typedInlined(Typer.scala:1768)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2715)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2756)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2941)
[#2] [error] dotty.tools.dotc.typer.Typer.$anonfun$34(Typer.scala:2156)
[#2] [error] dotty.tools.dotc.typer.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:225)
[#2] [error] dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2156)
[#2] [error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2670)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2755)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2847)
[#2] [error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2897)
[#2] [error] dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1025)
[#2] [error] dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1029)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2701)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2756)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedSelect(Inliner.scala:1492)
[#2] [error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2663)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2755)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2941)
[#2] [error] dotty.tools.dotc.typer.Applications.typedTypeApply(Applications.scala:1084)
[#2] [error] dotty.tools.dotc.typer.Applications.typedTypeApply$(Applications.scala:317)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedTypeApply(Inliner.scala:1549)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2712)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2756)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.Typer.typedInlined(Typer.scala:1768)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2715)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2756)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.Typer.typedInlined(Typer.scala:1768)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2715)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2756)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.ProtoTypes$FunProto.$anonfun$5(ProtoTypes.scala:428)
[#2] [error] dotty.tools.dotc.typer.ProtoTypes$FunProto.cacheTypedArg(ProtoTypes.scala:359)
[#2] [error] dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:429)
[#2] [error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:850)
[#2] [error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:850)
[#2] [error] dotty.tools.dotc.typer.Applications$Application.addTyped$1(Applications.scala:544)
[#2] [error] dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:609)
[#2] [error] dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:609)
[#2] [error] dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:609)
[#2] [error] dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:609)
[#2] [error] dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:447)
[#2] [error] dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:735)
[#2] [error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:849)
[#2] [error] dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1057)
[#2] [error] dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:317)
[#2] [error] dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:106)
[#2] [error] dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:904)
[#2] [error] dotty.tools.dotc.typer.Applications.realApply$5$$anonfun$4(Applications.scala:983)
[#2] [error] dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:2949)
[#2] [error] dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:994)
[#2] [error] dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1032)
[#2] [error] dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:317)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedApply(Inliner.scala:1535)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2693)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2756)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.Typer.typedInlined(Typer.scala:1768)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2715)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2756)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.Typer.typedInlined(Typer.scala:1768)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2715)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2756)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedSelect(Inliner.scala:1492)
[#2] [error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2663)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2755)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2941)
[#2] [error] dotty.tools.dotc.typer.Applications.typedTypeApply(Applications.scala:1084)
[#2] [error] dotty.tools.dotc.typer.Applications.typedTypeApply$(Applications.scala:317)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedTypeApply(Inliner.scala:1549)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2712)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2756)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.ProtoTypes$FunProto.$anonfun$5(ProtoTypes.scala:428)
[#2] [error] dotty.tools.dotc.typer.ProtoTypes$FunProto.cacheTypedArg(ProtoTypes.scala:359)
[#2] [error] dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:429)
[#2] [error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:850)
[#2] [error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:850)
[#2] [error] dotty.tools.dotc.typer.Applications$Application.addTyped$1(Applications.scala:544)
[#2] [error] dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:609)
[#2] [error] dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:447)
[#2] [error] dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:735)
[#2] [error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:849)
[#2] [error] dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1057)
[#2] [error] dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:317)
[#2] [error] dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:106)
[#2] [error] dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:904)
[#2] [error] dotty.tools.dotc.typer.Applications.realApply$5$$anonfun$4(Applications.scala:983)
[#2] [error] dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:2949)
[#2] [error] dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:994)
[#2] [error] dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1032)
[#2] [error] dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:317)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedApply(Inliner.scala:1535)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2693)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2756)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.ProtoTypes$FunProto.$anonfun$5(ProtoTypes.scala:428)
[#2] [error] dotty.tools.dotc.typer.ProtoTypes$FunProto.cacheTypedArg(ProtoTypes.scala:359)
[#2] [error] dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:429)
[#2] [error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:850)
[#2] [error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:850)
[#2] [error] dotty.tools.dotc.typer.Applications$Application.addTyped$1(Applications.scala:544)
[#2] [error] dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:609)
[#2] [error] dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:447)
[#2] [error] dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:735)
[#2] [error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:849)
[#2] [error] dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1057)
[#2] [error] dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:317)
[#2] [error] dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:106)
[#2] [error] dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:904)
[#2] [error] dotty.tools.dotc.typer.Applications.realApply$5$$anonfun$4(Applications.scala:983)
[#2] [error] dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:2949)
[#2] [error] dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:994)
[#2] [error] dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1032)
[#2] [error] dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:317)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedApply(Inliner.scala:1535)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2693)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2756)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.Typer.typedInlined(Typer.scala:1768)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2715)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2756)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2818)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2941)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedApply(Inliner.scala:1540)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2693)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2756)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedTyped(ReTyper.scala:62)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2698)
[#2] [error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2756)
[#2] [error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[#2] [error] dotty.tools.dotc.typer.Inliner$InlineTyper.typedUnadapted(Inliner.scala:1622)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2821)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2818)
[#2] [error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2825)
[#2] [error] dotty.tools.dotc.typer.Inliner.inlined(Inliner.scala:1016)
[#2] [error] dotty.tools.dotc.typer.Inliner$.inlineCall(Inliner.scala:153)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:86)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1420)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform$$anonfun$1(Trees.scala:1490)
[#2] [error] scala.collection.immutable.List.mapConserve(List.scala:472)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1490)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1386)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1420)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1420)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform$$anonfun$1(Trees.scala:1490)
[#2] [error] scala.collection.immutable.List.mapConserve(List.scala:472)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1490)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1386)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1420)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform$$anonfun$1(Trees.scala:1490)
[#2] [error] scala.collection.immutable.List.mapConserve(List.scala:472)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1490)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1418)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1420)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1394)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:82)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform$$anonfun$1(Trees.scala:1490)
[#2] [error] scala.collection.immutable.List.mapConserve(List.scala:472)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1490)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1386)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1420)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1420)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1380)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform$$anonfun$1(Trees.scala:1490)
[#2] [error] scala.collection.immutable.List.mapConserve(List.scala:472)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1490)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1386)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1420)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1420)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform$$anonfun$1(Trees.scala:1490)
[#2] [error] scala.collection.immutable.List.mapConserve(List.scala:472)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1490)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1386)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1420)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1420)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform$$anonfun$1(Trees.scala:1490)
[#2] [error] scala.collection.immutable.List.mapConserve(List.scala:472)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1490)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1386)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1420)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1420)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1454)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:80)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.traverse$1(TreeMapWithImplicits.scala:53)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transformStats(TreeMapWithImplicits.scala:60)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:111)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:80)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1463)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:80)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.traverse$1(TreeMapWithImplicits.scala:53)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transformStats(TreeMapWithImplicits.scala:60)
[#2] [error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1474)
[#2] [error] dotty.tools.dotc.ast.TreeMapWithImplicits.transform(TreeMapWithImplicits.scala:120)
[#2] [error] dotty.tools.dotc.transform.Inlining$InliningTreeMap.transform(Inlining.scala:94)
[#2] [error] dotty.tools.dotc.transform.Inlining$$anon$2.transform(Inlining.scala:72)
[#2] [error] dotty.tools.dotc.transform.MacroTransform.run(MacroTransform.scala:21)
[#2] [error] dotty.tools.dotc.transform.Inlining.run(Inlining.scala:44)
[#2] [error] dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:303)
[#2] [error] scala.collection.immutable.List.map(List.scala:250)
[#2] [error] dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:304)
[#2] [error] dotty.tools.dotc.transform.Inlining.runOn(Inlining.scala:48)
[#2] [error] dotty.tools.dotc.Run.runPhases$4$$anonfun$4(Run.scala:205)
[#2] [error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[#2] [error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[#2] [error] scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
[#2] [error] dotty.tools.dotc.Run.runPhases$5(Run.scala:216)
[#2] [error] dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:224)
[#2] [error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
[#2] [error] dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:67)
[#2] [error] dotty.tools.dotc.Run.compileUnits(Run.scala:231)
[#2] [error] dotty.tools.dotc.Run.compileSources(Run.scala:166)
[#2] [error] dotty.tools.dotc.Run.compile(Run.scala:150)
[#2] [error] dotty.tools.dotc.Driver.doCompile(Driver.scala:39)
[#2] [error] dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:88)
[#2] [error] dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
[#2] [error] sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
[#2] [error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:192)
[#2] [error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
[#2] [error] sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:247)
[#2] [error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:182)
[#2] [error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:163)
[#2] [error] sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[#2] [error] sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:163)
[#2] [error] sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:210)
[#2] [error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:528)
[#2] [error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:528)
[#2] [error] sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:177)
[#2] [error] sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:175)
[#2] [error] sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:461)
[#2] [error] sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[#2] [error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[#2] [error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[#2] [error] sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:263)
[#2] [error] sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:416)
[#2] [error] sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:503)
[#2] [error] sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:403)
[#2] [error] sbt.internal.inc.Incremental$.apply(Incremental.scala:169)
[#2] [error] sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:528)
[#2] [error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:482)
[#2] [error] sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[#2] [error] sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:420)
[#2] [error] sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[#2] [error] mill.scalalib.worker.ZincWorkerImpl.compileInternal(ZincWorkerImpl.scala:544)
[#2] [error] mill.scalalib.worker.ZincWorkerImpl.$anonfun$compileMixed0$1(ZincWorkerImpl.scala:317)
[#2] [error] mill.api.FixSizedCache.withCachedValue(FixSizedCache.scala:66)
[#2] [error] mill.scalalib.worker.ZincWorkerImpl.withCompilers(ZincWorkerImpl.scala:399)
[#2] [error] mill.scalalib.worker.ZincWorkerImpl.compileMixed0(ZincWorkerImpl.scala:316)
[#2] [error] mill.scalalib.worker.ZincWorkerImpl.compileMixed(ZincWorkerImpl.scala:284)
[#2] [error] mill.scalalib.ScalaModule.$anonfun$compile$2(ScalaModule.scala:195)
[#2] [error] mill.define.Task$TraverseCtx.evaluate(Task.scala:376)
[#2] [error] mill.eval.Evaluator.$anonfun$evaluateGroup$13(Evaluator.scala:505)
[#2] [error] scala.util.DynamicVariable.withValue(DynamicVariable.scala:59)
[#2] [error] scala.Console$.withErr(Console.scala:193)
[#2] [error] mill.eval.Evaluator.$anonfun$evaluateGroup$12(Evaluator.scala:505)
[#2] [error] scala.util.DynamicVariable.withValue(DynamicVariable.scala:59)
[#2] [error] scala.Console$.withOut(Console.scala:164)
[#2] [error] mill.eval.Evaluator.$anonfun$evaluateGroup$11(Evaluator.scala:504)
[#2] [error] scala.util.DynamicVariable.withValue(DynamicVariable.scala:59)
[#2] [error] scala.Console$.withIn(Console.scala:227)
[#2] [error] mill.eval.Evaluator.$anonfun$evaluateGroup$8(Evaluator.scala:503)
[#2] [error] mill.eval.Evaluator.$anonfun$evaluateGroup$8$adapted(Evaluator.scala:464)
[#2] [error] scala.collection.immutable.Vector.foreach(Vector.scala:1856)
[#2] [error] mill.eval.Evaluator.evaluateGroup(Evaluator.scala:464)
[#2] [error] mill.eval.Evaluator.$anonfun$evaluateGroupCached$13(Evaluator.scala:359)
[#2] [error] scala.util.DynamicVariable.withValue(DynamicVariable.scala:59)
[#2] [error] mill.eval.Evaluator.evaluateGroupCached(Evaluator.scala:350)
[#2] [error] mill.eval.Evaluator.$anonfun$parallelEvaluate$2(Evaluator.scala:227)
[#2] [error] scala.concurrent.impl.Promise$Transformation.run(Promise.scala:467)
[#2] [error] java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[#2] [error] java.util.concurrent.FutureTask.run(FutureTask.java:266)
[#2] [error] java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[#2] [error] java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[#2] [error] java.lang.Thread.run(Thread.java:748)
```